tutorial/numpy-functions.md

```diff
@@ -0,0 +1,300 @@
+# Supported Numpy function by Pyccel
+
+In Pyccel we try to support the most used Numpy functions by developers. here is some of them:
+
+## [Norm](https://numpy.org/doc/stable/reference/generated/numpy.linalg.norm.html)
+
+- Supported parameters:
+
+    x: array_like
+       Input array. If axis is None, x must be 1-D or 2-D, unless ord is None.
+       If both axis and ord are None, the 2-norm of x.ravel will be returned.
+
+    axis: {None, int, 2-tuple of ints}, optional.
+         If axis is an integer, it specifies the axis of x along which to compute the vector norms.
+         If axis is a 2-tuple, it specifies the axes that hold 2-D matrices, and the matrix norms of
+         these matrices are computed. If axis is None then either a vector norm (when x is 1-D) or a
+         matrix norm (when x is 2-D) is returned. The default is None. New in version 1.8.0.
+
+- Supported languages: Fortran
+
+- python code:
+
+    ```python
+    from numpy.linalg import norm
+    from numpy import array
+    arr1 = array([1,2,3,4])
+    nrm = norm(arr1)
+    print(nrm)
+
+    arr2 = array([[1,2,3,4],[4,3,2,1]])
+    nrm2 = norm(arr2, axis=1)
+    print(nrm)
+    ```
+
+- fortran equivalent:
+
+    ```fortran
+    program prog_test_norm
+
+    use, intrinsic :: ISO_C_BINDING
+
+    implicit none
+
+    integer(C_INT64_T), allocatable :: arr1(:)
+    real(C_DOUBLE) :: nrm
+    integer(C_INT64_T), allocatable :: arr2(:,:)
+    real(C_DOUBLE), allocatable :: nrm2(:)
+
+    allocate(arr1(0:4_C_INT64_T - 1_C_INT64_T))
+    arr1 = [1_C_INT64_T, 2_C_INT64_T, 3_C_INT64_T, 4_C_INT64_T]
+    nrm = Norm2(Real(arr1, C_DOUBLE))
+    print *, nrm
+    allocate(arr2(0:4_C_INT64_T - 1_C_INT64_T, 0:2_C_INT64_T - 1_C_INT64_T))
+    arr2 = reshape([[1_C_INT64_T, 2_C_INT64_T, 3_C_INT64_T, 4_C_INT64_T], [ &
+          4_C_INT64_T, 3_C_INT64_T, 2_C_INT64_T, 1_C_INT64_T]], [ &
+          4_C_INT64_T, 2_C_INT64_T])
+    allocate(nrm2(0:2_C_INT64_T - 1_C_INT64_T))
+    nrm2 = Norm2(Real(arr2, C_DOUBLE),2_C_INT64_T - 1_C_INT64_T)
+    print *, nrm
+
+    end program prog_test_norm
+    ```
+
+## [Real](https://numpy.org/doc/stable/reference/generated/numpy.real.html) and [imag](https://numpy.org/doc/stable/reference/generated/numpy.imag.html) functions
+
+- Supported languages: C (scalars only), fortran
```

Comment:
`real` and `imag` are supported for arrays in C. Your example does not translate, but that is due to the `print` not the calls to `real` or `imag`

Reply:
the print in the generated C code will print something like arrays addresses since we don’t print arrays in C with the right way, actually we just pass the array to `printf` right now, so `printf` will just cast the argument to the given type and print it, mostly for arrays or pointers in general printf will print the addresses.

Reply:
so for now we can just put generated C code without `printf`, if we want to make it make more sense.

Reply:
@mostafaMamoni I fixed printf for you. See PR #833. Hopefully this should make things easier for you :)

Reply:
ok, thanks Emily.